File: livingdoc-fixture-api/src/test/java/org/livingdoc/fixture/api/converter/common/LocalDateConverterTest.java
package org.livingdoc.fixture.api.converter.common;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

import org.junit.jupiter.api.Test;
import org.livingdoc.fixture.api.converter.ConversionException;


class LocalDateConverterTest {

    LocalDateConverter cut = new LocalDateConverter();

    @Test
    void valuesAreConvertedCorrectly() {
        LocalDate now = LocalDate.parse("2017-05-05");
        String formatted = now.format(DateTimeFormatter.ISO_DATE);
        assertThat(cut.convert(formatted)).isEqualTo(now);
    }

    @Test
    void illegalValueThrowsConversionException() {
        assertThrows(ConversionException.class, () -> cut.convert("no Date"));
    }

}

File: livingdoc-fixture-converter/src/test/kotlin/org/livingdoc/fixture/converter/time/LocalDateConverterTest.kt
package org.livingdoc.fixture.converter.time

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDate
import java.time.LocalDate.parse


internal class LocalDateConverterTest : TemporalConverterContract<LocalDate>() {

    override val cut = LocalDateConverter()

    override val validInputVariations = mapOf(
            "2017-05-12" to parse("2017-05-12")
    )

    override val defaultFormatValue = "2017-05-12" to parse("2017-05-12")

    override val customFormat = "dd.MM.uuuu"
    override val customFormatValue = "12.05.2017" to parse("2017-05-12")
    override val malformedCustomFormat = "dd.MM.uuuu V"

    @Test
    fun `converter can converted to LocalDate`() {
        assertThat(cut.canConvertTo(LocalDate::class.java)).isTrue()
    }

}
